tests/conftest.py

```diff
@@ -308,6 +312,48 @@ def _generate_upf_data(element):
     return _generate_upf_data
 
 
+@pytest.fixture(scope='session')
+def generate_singlefile_data():
```

Comment:
Well but it is not generic is it? Because it will always return a `SinglefileData` with content that is specifically that of a core_wfc file? Why would any other test use it that would require other content? But even more so, creating a `SinglefileData` is as easy as `SinglefileData(io.StringIO('some content'))` so do we really even need a fixture for this?